### PR TITLE
Feature/shape demo test

### PR DESCRIPTION
**Pull request by lemunozm**
Added a new integration test using the shapes demo commands.

This pull request requires of https://github.com/eProsima/Micro-XRCE-DDS-Client/pull/49 to work fine.

### CODE REVIEW

File: test/shapes_demo/ShapesDemo.cpp [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    std::vector<std::string> commands_;
+};
+
+/* ############################################## TESTS ##################################################### */
+
+INSTANTIATE_TEST_CASE_P(Transport, ShapesDemoTest, ::testing::Values(UDP_TRANSPORT, TCP_TRANSPORT), ::testing::PrintToStringParamName());
+
+TEST_P(ShapesDemoTest, CreateSession)
+{
+    commands_.push_back("create_session");
```
Comment (BorjaOuterelo) [inline test/shapes_demo/ShapesDemo.cpp]: Yo can use `std::vector::emplace_back` avoiding a temporary.
This applies to all the TEST here.

File: test/shapes_demo/ShapesDemo.cpp [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+        agent_->stop();
+        ASSERT_EQ(0, shape_demo_app_result);
+    }
+
+protected:
+    eprosima::uxr::Server* init_agent(uint16_t port)
+    {
+        eprosima::uxr::Server* agent;
+        switch(transport_)
```
Comment (BorjaOuterelo) [inline test/shapes_demo/ShapesDemo.cpp]: Add the `default` label.

Comment (lemunozm) [reply test/shapes_demo/ShapesDemo.cpp]: Done

File: test/shapes_demo/ShapesDemo.cpp (hunk 1 above)
Comment (lemunozm) [reply test/shapes_demo/ShapesDemo.cpp]: Done

**Pull request merged**